$MirOS: ports/devel/cvs/patches/patch-src_lock_c,v 1.1 2006/10/02 05:59:17 tg Exp $
--- src/lock.c.orig	Sat Mar 10 23:14:20 2007
+++ src/lock.c	Sat Mar 10 23:14:14 2007
@@ -71,8 +71,8 @@
 
 #include "cvs.h"
 
+__RCSID("$MirOS: src/gnu/usr.bin/cvs/src/lock.c,v 1.3 2006/03/27 11:36:22 tg Exp $");
 
-
 struct lock {
     /* This is the directory in which we may have a lock named by the
        readlock variable, a lock named by the writelock variable, and/or
@@ -1318,6 +1318,10 @@ internal_lock (struct lock *lock, const 
     assert (!lock->repository);
     lock->repository = Xasprintf ("%s/%s", xrepository, CVSROOTADM);
     lock->free_repository = true;
+
+    /* do nothing if we know it fails anyway */
+    if (readonlyfs)
+      return 0;
 
     /* get the lock dir for our own */
     if (set_lock (lock, 1) != L_OK)
